docs/website/contents/for-developers/HardForkCombinatorRollbackMonotonicityProposal.md

```diff
@@ -0,0 +1,285 @@
+
+- QSR - Queries are Sensitivity to Rollback - same as DSR but for HFC-related Node-To-Client queries instead of for validity of blocks: Sensitive vs Insensitive
+
+Note the following relations.
+
+- ZEE=Allowed implies ZBE=Allowed.
+- DSR=Insensitive implies QSR=Insensitive.
+
+## Table for the Current Cardano HFC
+
+The following table and associated notes classifies the current HFC implementation in terms of these dimensions.
+
+| DST | CCT | PIL | DSR | ZEE | ZBE | QSR |
+| --- | --- | --- | --- | --- | --- | --- |
+| I   | A   | P 1 | I   | 2   | 3   | I   |
+
+- (1) It's Possible, but only if there's a Chain Growth violation at a particular time during the epoch.
+
+- (2) The current HFC's behavior is inconsistent without making further assumptions.
+  Overall, ZEE is allowed, but it might cause some unexpected behavior.
+  In particular, forecasts always anticipate that there is at least one safe zone of slots in an era (which is not true for eras with zero epochs).
+  That mismatch could manifest as incorrect time-slot translations eg (within ledger rules execution, but not within Node-To-Client query reponses).
+
+- (3) ZBE is allowed when ticking a ledger state; the HFC will invoke multiple translation functions in sequence.
+  But cross-era forecasting cannot skip eras, so no headers after an epoch with zero blocks could be validated, and so blocks could not progatate in the net.
+  Moreover, today's node would not even mint blocks, since the leadership check (at least currently) relies on forecasting.
+
+## Table for an Particular Hypothetical Cardano HFC
+
+The following table and associated notes classifies an hypothetical HFC implementation in terms of these dimensions.
+Nicolas Frisby anticipates this implementation would be easier to understand, less likely to surprise users, more general (ie place fewer requirements on the block types), and not require excessive code changes.
+
+| DST | CCT | PIL | DSR | ZSE | ZBE | QSR |
+| --- | --- | --- | --- | --- | --- | --- |
+| S   | D   | I   | S 2 | D   | D 1 | I 3 |
+
+- (1) ZBE would be achievable for a prefix of eras when constructing the initial HFC ledger state (in the `ProtocolInfo` passed to the Consensus layer), but otherwise the HFC as a block type does not need to support ZBE at all.
+
+- (2) Esgen points out that perhaps DSR could/should still be Insensitive, and that wouldn't obviously lose too much simplicity.
+
+- (3) Having QSR=Insensitive despite DSR=Sensitive requires an additional mechanism.
+  In this design point, that is anticipated to be constraining these queries to only be answered via an immutable ledger state.
+
+## Disclaimer
+
+There may be other points in the design space worth considering.
+(Or even other dimensions, of course.)
+In particular, DST=Insensitive and PIL=Impossible could be achieved by changing the Cardano ledger's governance to match the existing Cardano HFC logic.
```

Comment:
I can't imagine/understand why this would be the case. Would it be possible to illustrate/expand on this explanation?

Reply:
I had intended the next sentence to do so. So I suppose we should expand that one.